js/main.js

```diff
@@ -56,36 +58,43 @@ window.addEventListener('DOMContentLoaded', function (event) {
   for (var i = 0; i < data.entries.length; i++) {
     $ul.appendChild(renderEntry(data.entries[i]));
   }
-  var JSONDataView = localStorage.getItem('code-journal-data-view');
-  if (JSONDataView === 'entries') {
-    $dataViewEntryForm.className = 'container data-form hidden';
-    $dataViewEntries.className = 'container data-entries';
-  } else {
-    $dataViewEntryForm.className = 'container data-form';
-    $dataViewEntries.className = 'container data-entries hidden';
+  for (var j = 0; j < $allView.length; j++) {
+    if ($allView[j].getAttribute('data-view') === data.view) {
+      $allView[j].className = 'container view';
+    } else {
+      $allView[j].className = 'container view hidden';
+    }
   }
 });
 
 var $nav = document.querySelector('nav');
-var $dataViewEntryForm = document.querySelector('.data-form');
-var $dataViewEntries = document.querySelector('.data-entries');
+var $allView = document.querySelectorAll('.view');
+
 $nav.addEventListener('click', function (event) {
   if (event.target.matches('.entries-link')) {
-    $dataViewEntryForm.className = 'container data-form hidden';
-    $dataViewEntries.className = 'container data-entries';
+    for (var i = 0; i < $allView.length; i++) {
+      if ($allView[i].getAttribute('data-view') === 'entries') {
+        $allView[i].className = 'container view';
+      } else {
+        $allView[i].className = 'container view hidden';
+      }
+    }
   }
-  var $entriesDataView = $dataViewEntries.getAttribute('data-view');
-  localStorage.setItem('code-journal-data-view', $entriesDataView);
+  data.view = 'entries';
 });
 
 var $newButton = document.querySelector('.new-button');
 $newButton.addEventListener('click', function (event) {
   if (event.target.matches('.new-button')) {
-    $dataViewEntryForm.className = 'container data-form';
-    $dataViewEntries.className = 'container data-entries hidden';
+    for (var i = 0; i < $allView.length; i++) {
+      if ($allView[i].getAttribute('data-view') === 'entry-form') {
+        $allView[i].className = 'container view';
+      } else {
+        $allView[i].className = 'container view hidden';
+      }
+    }
```

Comment:
You've repeated this exact same behavior with only minor adjustments three times. You should make a reusable function that has a parameter of a string which is the name of the view you want to show, which loops over your list of views and shows the view with a matching name and hides all other views that don't match.

Reply:
🤔